test/gtest/ucp/test_ucp_proto.cc

```diff
@@ -409,13 +464,25 @@ class test_proto_perf : public ucp_test {
         }
     }
 
+private:
+    static perf_ptr_t make_perf(ucp_proto_perf_t *perf)
+    {
+        return perf_ptr_t(perf, ucp_proto_perf_destroy);
+    }
```

Comment:
Minor:
```suggestion
    static perf_ptr_t make_perf_shared_ptr(ucp_proto_perf_t *perf)
    {
        return {perf, ucp_proto_perf_destroy};
    }
```

1. Extend the name since the function doesn't make any `perf` but just creates shared ptr on top of it (`make_perf_ptr` or `make_perf_shared` seems also good options for me)